front/src/reducers/main/mainReducer.spec.ts

```diff
@@ -0,0 +1,170 @@
+  MainState,
+} from 'reducers/main';
+
+const createStore = (initialStateExtra: MainState) =>
+  createStoreWithoutMiddleware({
+    main: initialStateExtra,
+  });
+
+describe('mainReducer', () => {
+  test('should return the initial state', () => {
+    const store = createStore(mainInitialState);
+    const mainState = store.getState().main;
+    expect(mainState).toEqual(mainInitialState);
+  });
+
+  test('should handle setLoading', () => {
+    const store = createStore(mainInitialState);
+    const action = setLoading();
+    store.dispatch(action);
+    const mainState = store.getState().main;
+    expect(mainState).toEqual({
+      loading: 1,
+      notifications: [],
+      lastInterfaceVersion: '',
+    });
+  });
+
+  test('should handle setSuccess', () => {
+    const store = createStore(mainInitialState);
+    const action = setSuccess({
+      type: 'success',
+      title: 'Test title',
+      text: 'Test text',
+      date: new Date(),
+    });
+    store.dispatch(action);
+    const mainState = store.getState().main;
+    expect(mainState).toEqual({
+      loading: 0,
+      notifications: [
+        {
+          type: 'success',
+          title: 'Test title',
+          text: 'Test text',
+          date: action.payload?.date,
+        },
+      ],
+      lastInterfaceVersion: '',
+    });
+  });
+
+  test('should handle setWarning', () => {
+    const store = createStore(mainInitialState);
+    const action = setWarning({
+      type: 'warning',
+      title: 'Test title',
+      text: 'Test text',
+      date: new Date(),
+    });
+    store.dispatch(action);
+    const mainState = store.getState().main;
+    expect(mainState).toEqual({
+      loading: 0,
+      notifications: [
+        {
+          type: 'warning',
+          title: 'Test title',
+          text: 'Test text',
+          date: action.payload?.date,
+        },
+      ],
+      lastInterfaceVersion: '',
+    });
+  });
+
+  test('should handle setFailure', () => {
+    const store = createStore(mainInitialState);
+    const action = setFailure(new Error('Test error'));
+    store.dispatch(action);
+    const mainState = store.getState().main;
+    expect(mainState).toEqual({
+      loading: 0,
+      notifications: [
+        {
+          type: 'error',
+          title: 'Error',
+          text: 'Test error',
+          date: new Date(),
+        },
+      ],
+      lastInterfaceVersion: '',
+    });
+  });
+
+  test('should handle addNotification', () => {
+    const store = createStore(mainInitialState);
+    const action = addNotification({
+      type: 'warning',
+      title: 'Test title',
+      text: 'Test text',
+      date: new Date(),
+    });
+    store.dispatch(action);
+    const mainState = store.getState().main;
+    expect(mainState).toEqual({
+      loading: 0,
+      notifications: [
+        {
+          type: 'warning',
+          title: 'Test title',
+          text: 'Test text',
+          date: action.payload?.date,
```

Comment:
We should not need the ?
```suggestion
          date: action.payload.date,
```